lms/djangoapps/grades/subsection_grade.py

```diff
@@ -182,24 +182,34 @@ def _compute_block_score(
                     block,
                 )
 
+    @staticmethod
+    def _aggregated_score_from_model(grade_model, is_graded):
+        """
+        Helper method that returns `AggregatedScore` objects based on
+        the values in the given `grade_model`.  If the given model
+        has an associated override, the values from the override are
+        used instead.
+        """
+        score_type = 'graded' if is_graded else 'all'
+        grade_object = grade_model
+        if hasattr(grade_model, 'override'):
+            score_type = 'graded_override' if is_graded else 'all_override'
+            grade_object = grade_model.override
+        return AggregatedScore(
+            tw_earned=getattr(grade_object, 'earned_{}'.format(score_type)),
+            tw_possible=getattr(grade_object, 'possible_{}'.format(score_type)),
+            graded=is_graded,
+            first_attempted=grade_model.first_attempted,
+        )
+
 
 class ReadSubsectionGrade(NonZeroSubsectionGrade):
     """
     Class for Subsection grades that are read from the database.
     """
     def __init__(self, subsection, model, factory):
-        all_total = AggregatedScore(
-            tw_earned=model.earned_all,
-            tw_possible=model.possible_all,
-            graded=False,
-            first_attempted=model.first_attempted,
-        )
-        graded_total = AggregatedScore(
-            tw_earned=model.earned_graded,
-            tw_possible=model.possible_graded,
-            graded=True,
-            first_attempted=model.first_attempted,
-        )
+        all_total = self._aggregated_score_from_model(model, is_graded=False)
+        graded_total = self._aggregated_score_from_model(model, is_graded=True)
```

Comment:
Nit: not a blocker - but I found it confusing in reading this to understand when `_update_aggregated_scores_from_model` is to be called in some cases versus `_aggregated_score_from_model`.  Perhaps it's a matter of just renaming the helpers to clarify their distinctions.

Reply:
Good point, I've just eliminated `_update_aggregated_scores_from_model` as a helper method, since it's only 3 lines and only called in one place (I've left the doc-string as a comment where it's called).